src/async_impl/multipart.rs

```diff
@@ -244,6 +244,14 @@ impl Part {
         self.with_inner(move |inner| inner.file_name(filename))
     }
 
+    /// Sets custom headers for the part.
+    pub fn headers<T>(self, headers: T) -> Part
+    where
+        T: Into<HeaderMap>,
```

Comment:
The request builders just take `HeaderMap`, I'd lean towards keeping it the same. We can add this generic later to all of them if there's a good argument for doing so.

Reply:
Ok with me!